### PR TITLE
Set up continuous deployment for ide-extension

### PR DESCRIPTION
**Pull request by openscript**
### No scoped extension names
VSCodes Marketplace doesn't support scoped names like `@inlang/ide-extension`. `vsce` infers the name from `package.json`. This leads to: `ERROR  Invalid extension name '@inlang/ide-extension'`.

https://github.com/microsoft/vscode-vsce/issues/186

### VSCE adds .js to entry points

Because inlang are ESMs, but ide-extension cannot be ESM, we use CommonJS for it with `.cjs` extension to declare this. Prior `2.18.0` of `vsce` it added `.js` to the entrypoint and eventually searching for a `main.cjs.js` file. Luckily this is fixed in the most recent version.

https://github.com/microsoft/vscode-vsce/pull/825

### Monorepo 

`vsce` tries to package parent directories as well. This isn't necessary and leads to errors (naming collisions in node_modules), so they should be ignored inside `.vscodeignore`:

```
../../**
../**
```

### CODE REVIEW

File: source-code/ide-extension/package.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
-	"name": "@inlang/ide-extension",
+	"name": "inlang",
```
[the remark this answers is not in the document]
Comment (openscript) [reply source-code/ide-extension/package.json:2]: It was a mistake that I pushed this. There is a `script` which renames the name upon build.

File: source-code/ide-extension/scripts/preparePackage.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import("node:fs").then((fs) => {
```
Comment (samuelstroschein) [inline source-code/ide-extension/scripts/preparePackage.js:1]: why are you not using a regular import?

File: source-code/ide-extension/package.json (hunk 1 above)
Comment (samuelstroschein) [inline source-code/ide-extension/package.json]: the name is still inlang. it should be "vs-code-extension"

Comment (openscript) [reply source-code/ide-extension/package.json]: That is weird:
image

https://github.com/inlang/inlang/pull/446/commits/742ad605e4f42b102e18f540af0d494437afe7a5

Comment (samuelstroschein) [reply source-code/ide-extension/package.json]: Okay let's merge. Either it deplys or it doesn't

**Comment by samuelstroschein** [comment]
@openscript  Feel free to merge

**Pull request merged**